src/stravalib/tests/integration/test_client.py

```diff
@@ -885,6 +891,52 @@ def test_get_segment_effort(mock_strava_api, client):
     assert effort.max_heartrate == 170
 
 
+@pytest.mark.parametrize(
+    "params, warning_type, warning_message",
+    [
+        (
+            {"segment_id": 2345, "athlete_id": 12345},
+            DeprecationWarning,
+            'The "athlete_id" parameter is unsupported',
+        ),
+        (
+            {"segment_id": 2345, "limit": 10},
+            DeprecationWarning,
+            'The "limit" parameter is deprecated',
+        ),
+    ],
+)
+def test_get_segment_efforts_warnings(
+    params, warning_type, warning_message, mock_strava_api, client
+):
+    """Test that if provided with deprecated params, the user receives a
+    warning."""
+
+    mock_strava_api.get(
+        "/segment_efforts",
+        response_update={"name": "Best Run Ever"},
+        n_results=1,
+    )
+
+    with pytest.warns(warning_type, match=warning_message):
+        a = client.get_segment_efforts(**params)
+        next(a)
+
+
+def test_get_segment_efforts(client, mock_strava_api):
+    """Test that endpoint returns data as expected."""
+    mock_strava_api.get("/segment_efforts", n_results=4)
+
+    efforts = client.get_segment_efforts(segment_id=2345, athlete_id=12345)
+
+    for index, effort in enumerate(efforts):
+        if index == 0:
+            name = effort.name
+
+    assert index + 1 == 4
+    assert name == "Alpe d'Huez"
```

Comment:
```suggestion
    efforts = list(client.get_segment_efforts(segment_id=2345, athlete_id=12345))

    assert len(efforts) == 4
    assert efforts[0].name == "Alpe d'Huez"
```
I tried to make the test assertions more explicit here, but the test logic was already sound!

Reply:
Yay! thanks!! i see - .name doesn't really relate to the efforts endpoint so this is much better. merging now.